olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -392,8 +404,11 @@ bool DefaultCacheImpl::Remove(const std::string& key) {
 
   if (mutable_cache_) {
     uint64_t removed_data_size = 0;
-    PurgeDiskItem(key, *mutable_cache_, removed_data_size);
-
+    if (!PurgeDiskItem(key, *mutable_cache_, removed_data_size)) {
+      OLP_SDK_LOG_ERROR_F(kLogTag, "Remove() failed to purge item, key='%s'",
+                          key.c_str());
+      return false;
+    }
     mutable_cache_data_size_ -= removed_data_size;
```

Comment:
In case we removed something but exited on L410, we do not modify the mutable_cache_data_size_

Reply:
Good catch! Fixed